home2.2/Main.java

```diff
@@ -0,0 +1,190 @@
+import java.io.*;
+
+class Pixel {
+    int r, g, b;
+    Pixel(int r, int g, int b) {
+        this.r = r;
+        this.g = g;
+        this.b = b;
+    }
+}
+
+class ImageProcessor {
+    private Pixel[][] image;
+    private Pixel favorite;
+    private Pixel unfavorite; // New unfavorite color
+    private int rows;
+    private int cols;
+
+    public ImageProcessor(int rows, int cols) {
+        this.rows = rows;
+        this.cols = cols;
+        image = new Pixel[rows][cols];
+    }
+
+    public void setFavoriteColor(int r, int g, int b) {
+        if (isValidColor(r, g, b)) {
+            favorite = new Pixel(r, g, b);
+        } else {
+            System.err.println("Invalid favorite color");
+            System.exit(1);
+        }
+    }
+
+    public void setUnfavoriteColor(int r, int g, int b) {
+        if (isValidColor(r, g, b)) {
+            unfavorite = new Pixel(r, g, b);
+        } else {
+            System.err.println("Invalid unfavorite color");
+            System.exit(1);
+        }
+    }
+
+    private boolean isValidColor(int r, int g, int b) {
+        return (r >= 0 && r <= 255) && (g >= 0 && g <= 255) && (b >= 0 && b <= 255);
+    }
+
+    private void processImage() {
+        for (int i = 0; i < rows; i++) {
+            for (int j = 0; j < cols; j++) {
+                if (unfavorite != null && image[i][j].r == unfavorite.r && image[i][j].g == unfavorite.g && image[i][j].b == unfavorite.b) {
```

Comment:
I hope we should override == operator for pixel to simplify the code to image[i][j] == unfavorite and so on
```
class Pixel {
...
@Override
 public boolean equals(Object o) {
```